emoji/src/main/java/com/vanniktech/emoji/EmojiImageView.java

```diff
@@ -66,9 +71,65 @@ public EmojiImageView(final Context context, final AttributeSet attrs) {
     }
   }
 
-  public void setHasVariants(final boolean hasVariants) {
-    this.hasVariants = hasVariants;
+  public void setEmoji(final Emoji emoji, final OnEmojiClickedListener clickListener, final OnEmojiLongClickedListener longClickListener) {
+    setImageDrawable(null);
+
+    setOnClickListener(new View.OnClickListener() {
+      @Override public void onClick(final View v) {
+        if (clickListener != null) {
+          clickListener.onEmojiClicked(emoji);
+        }
+      }
+    });
+
+    final Emoji baseEmoji = emoji.getBase();
 
-    invalidate();
+    hasVariants = baseEmoji.hasVariants();
+
+    if (hasVariants) {
+      setOnLongClickListener(new View.OnLongClickListener() {
+        @Override public boolean onLongClick(final View v) {
+          if (longClickListener != null) {
+            longClickListener.onEmojiLongClicked(v, emoji);
+
+            return true;
+          }
+
+          return false;
+        }
+      });
+    } else {
+      setOnLongClickListener(null);
+    }
+
+    ImageLoadingTask task = (ImageLoadingTask) getTag();
```

Comment:
I think the Task is only necessary when at once trying to display all the Emojis. I don't think we need it to just switch the variant (We know for sure that you can only do this one by one). I think I'll just end up calling `setImageResource()` and also avoid the `setDrawable(null)` call which should get rid of the flickering.